src/deploy/hosting/convertConfig.js

```diff
@@ -1,4 +1,25 @@
 const _ = require("lodash");
+const { FirebaseError } = require("../../error");
+
+/**
+ * makePattern contains the logic for extracting exactly one glob/regexp from
+ * a Hosting rewrite/redirect/header specification
+ */
+function makePattern(type, spec) {
```

Comment:
Maybe call this `extractPattern` instead of `makePattern`?

Reply:
Good call.